packages/voting-contract/src/ballots.rs

```diff
@@ -0,0 +1,74 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use cosmwasm_std::{Addr, Storage};
+use cw_storage_plus::{Index, IndexList, IndexedMap, MultiIndex};
+use tg3::Vote;
+
+use crate::ContractError;
+
+// we cast a ballot with our chosen vote and a given points
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+pub struct Ballot {
+    pub voter: Addr,
+    pub proposal_id: u64,
+    pub points: u64,
+    pub vote: Vote,
+}
+
+pub struct BallotIndexes<'a> {
+    // This PrimaryKey allows quering over all proposal ids for given voter address
+    pub voter: MultiIndex<'a, Addr, Ballot, (u64, Addr)>,
+}
+
+impl<'a> IndexList<Ballot> for BallotIndexes<'a> {
+    fn get_indexes(&'_ self) -> Box<dyn Iterator<Item = &'_ dyn Index<Ballot>> + '_> {
+        let v: Vec<&dyn Index<Ballot>> = vec![&self.voter];
+        Box::new(v.into_iter())
+    }
+}
+
+pub struct Ballots<'a> {
+    pub ballots: IndexedMap<'a, (u64, &'a Addr), Ballot, BallotIndexes<'a>>,
+}
+
+impl<'a> Ballots<'a> {
+    pub fn new(storage_key: &'a str, release_subkey: &'a str) -> Self {
+        let indexes = BallotIndexes {
+            voter: MultiIndex::new(|ballot| ballot.voter.clone(), storage_key, release_subkey),
```

Comment:
Here, if we go ahead and change the index function signature for `MultiIndex`, we could instead have
```suggestion
            voter: MultiIndex::new(|(pk, _ballot)| pk.1, storage_key, release_subkey),
```